ndcube/global_coords/global_coords.py

```diff
@@ -0,0 +1,66 @@
+from collections.abc import Mapping
+
+from ndcube import NDCube
+
+__all__ = ['GlobalCoords']
+
+
+class GlobalCoords(Mapping):
+    """
+    A structured representation of coordinate information applicable to a whole NDCube.
+    """
+    def __init__(self, NDCube):
+        super().__init__()
+
+        # Set values using the setters for validation
+        self.mapping = {}
+
+    def add(self, name, physical_type, coords):
+        """
+        Add a new coordinate to the collection.
+        """
+        self.mapping[name] = [name, physical_type, coords]
```

Comment:
Putting name in the list here seems pointless, given it's the key of the dict.

Reply:
Sure

Reply:
But I thought that is what @DanRyanIrish wanted

Reply:
I don't know, it seems redundant.

Reply:
Yeah, changed the LHS to `self._internal_coords[str(name)]` for defining the new dict entry instead